precompiles/ArbGasInfo.go

```diff
@@ -63,7 +64,10 @@ func (con ArbGasInfo) GetPricesInArbGasWithAggregator(c ctx, evm mech, aggregato
 
 	// aggregators compress calldata, so we must estimate accordingly
 	weiForL1Calldata := arbmath.BigMulByUint(l1GasPrice, params.TxDataNonZeroGasEIP2028)
-	gasForL1Calldata := arbmath.BigDiv(weiForL1Calldata, l2GasPrice)
+	gasForL1Calldata := common.Big0
+	if l2GasPrice.Sign() > 0 {
+		gasForL1Calldata = arbmath.BigDiv(weiForL1Calldata, l2GasPrice)
+	}
```

Comment:
Rather than doing this can we get the current base fee from state so that we can give an accurate amount of gas used to what will be used when executing

Reply:
I'm not sure what you mean. Getting the L2 gas price from the evm via `evm.Context.BaseFee` should be accurate. I made the change here just to harden the system a bit against the base fee being zero, which should never happen in practice but previously panicked as found by the fuzzer.

Reply:
Ah, I was thinking about in gas estimation mode where IIRC the base fee is setable? I forget what the behavior is when 0 is passed for the price

Reply:
In gas estimation mode you can set the gas fee cap of the message to zero, which is different